api/src/main/java/au/org/aodn/nrmn/restapi/service/SurveyIngestionService.java

```diff
@@ -100,7 +101,7 @@ public Survey getSurvey(Program program, OptionalDouble visAvg, StagedRowFormatt
 
         Optional<Survey> existingSurvey = surveyRepository.findOne(Example.of(survey));
 
-        return existingSurvey.orElseGet(() -> surveyRepository.save(
+        var returningSurvey = existingSurvey.orElseGet(() -> surveyRepository.save(
```

Comment:
You already do a save here with incorrect data, because surveyRepository.save () called.

Can you move the distanceBetween before this line and then the function call become  

.longitude(distance < 10d ? null : stagedRow.getLongitude())

Reply:
oh I miss that small `save` word and reckoned that part is just to build a survey. Thank you for reminding.

Reply:
Will try to do some slight change about the format.

Reply:
@utas-raymondng Hello Raymond. I have fixed that problem. Could you help me check the PR again?